tools/project_manager/show_word_context.php

```diff
@@ -40,103 +41,94 @@
 
 $wordInstances =  get_integer_param($_GET, 'wordInstances', 20, 0, MAX_WORD_INSTANCES);
 
-// $frame determines which frame we're operating from
-// 'master' - we're the master frame
-//  'left'  - we're the left frame with the text
-// 'right'  - we're the right frame for the image
-$frame = get_enumerated_param($_GET, 'frame', 'master', array('master', 'left', 'right'));
-
-if($frame=="master") {
-    slim_header_frameset(_("Word Context"));
-    if($layout == LAYOUT_HORIZ) $frameSpec='rows="30%,70%"';
-    else $frameSpec='cols="30%,70%"';
-?>
-<frameset <?php echo $frameSpec; ?>>
-<frame name="worddetailframe" src="show_word_context.php?projectid=<?php echo $projectid; ?>&amp;word=<?php echo $encWord; ?>&amp;wordInstances=<?php echo $wordInstances; ?>&amp;frame=left">
-<frame name="imageframe" src="show_word_context.php?projectid=<?php echo $projectid; ?>&amp;word=<?php echo $encWord; ?>&amp;wordInstances=<?php echo $wordInstances; ?>&amp;frame=right">
-</frameset>
-<noframes>
-<?php echo _("Your browser currently does not display frames!"); ?>
-</noframes>
-<?php
-    exit;
+$details = json_encode([
+    "layout" => $layout,
+    "projectid" => $projectid,
+]);
+
+$header_args = [
+    "js_files" => [
+        "$code_url/scripts/splitControl.js",
+        "$code_url/scripts/page_browse.js",
+        "./show_word_context.js",
+    ],
+    "js_data" => get_proofreading_interface_data_js() . "
+        var showWordContext = $details;",
+
+    "body_attributes" => 'class="no-margin overflow-hidden" style="height: 100vh; width: 100vw"',
+];
+
+slim_header(_("Word Context"), $header_args);
+echo "<div id='show_word_context_container' style='flex: auto;width: 100%;height: 100%'>";
+echo "<div class='overflow-auto'>";
+echo "<div style='padding: 0.5em'>";
+
+$project_name = get_project_name($projectid);
+// TRANSLATORS: %1$s is a word, %2$s is the project name.
+echo "<h2>", sprintf(_("Context for '%1\$s' in %2\$s"), $word, $project_name), "</h2>";
```

Comment:
would it make sense to use those multiline strings in php? https://phpf1.com/tutorial/php-heredoc-syntax.html

Reply:
I haven't had time to look at the code yet, but be aware that (unless something has changed in the last year or so) HEREDOC strings are not translatable.